generators/parser_test.go

```diff
@@ -0,0 +1,34 @@
+package generators
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestParse(t *testing.T) {
+	file, err := os.Open("demo.yaml")
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+
+	info, err := file.Stat()
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+
+	content := make([]byte, info.Size())
+	_, err = file.Read(content)
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+
+	var buf bytes.Buffer
+	buf.Write(content)
+	ret, err := Parse(&buf)
+	if err != nil {
+		t.Errorf("Error from 'Parse': %v", err)
+	}
+	fmt.Println(ret.SpecGroup.Spec.AccessModes[0])
```

Comment:
This by itself isn't a good test, since it won't actually fail if the content isn't what we expect. I don't know off the top of my head what the behavior of this parse library is, but many json/yaml libraries will ignore extra / unknown fields so long as the content is well-formed. If this is the case, I could pass an unrelated yaml file to this method and the test would pass, even though the resulting struct may be empty.

Let's verify the actual struct content. https://pkg.go.dev/github.com/google/go-cmp/cmp?tab=doc is a great library for comparing complex structs.

Also feel free to use a simpler struct if you want - we're likely going to be modifying the message entirely next.